### DISCUSSION TITLE
How to enable and disable tracing in a running application

### DISCUSSION THREAD

**Discussion by jaisonsteephen** [open-telemetry/community#738] [Community Discussions]
We can disable tracing with code(not by auto instrumenting), by setting **NewNoopTracerProvider** in Go. And later can enable by setting a NewTracerProvider
Is there similar options in other languages (Java, Python, Node), where **auto instrumentation** is done for http & gRPC calls.
I am more interested (welcoming other methods too) in something like adding new values(like NewNoopTracerProvider)  to environment variables, so that **auto instrumentation will take care remaining**.

**Comment by SergeyKanzhelev** [comment]
One way to find it is to explore registry. For example for Node: https://opentelemetry.io/registry/?s=http&component=instrumentation&language=js#

If you have question to specific instrumentation, it would be best if you'd ask in corresponding repository

**Reply by jaisonsteephen** [reply]
I think, assigning **NoopTrace**  is **more generic to opentelemetry**, not specific to any specific insrumentation
So you meant to say that, there is no such way available in Opentelemetry to enable/disable  tracing **if we are using autoinstrumentation** ( but need registry, which extends opentelemetry )
